homeassistant/components/geo_location/usgs_earthquake_hazards_program_feed.py

```diff
@@ -0,0 +1,269 @@
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect, dispatcher_send)
+from homeassistant.helpers.event import track_time_interval
+
+REQUIREMENTS = ['geojson_client==0.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_ALERT = 'alert'
+ATTR_EXTERNAL_ID = 'external_id'
+ATTR_MAGNITUDE = 'magnitude'
+ATTR_PLACE = 'place'
+ATTR_STATUS = 'status'
+ATTR_TIME = 'time'
+ATTR_TYPE = 'type'
+ATTR_UPDATED = 'updated'
+
+CONF_FEED_TYPE = 'feed_type'
+CONF_MINIMUM_MAGNITUDE = 'minimum_magnitude'
+
+DEFAULT_MINIMUM_MAGNITUDE = 0.0
+DEFAULT_RADIUS_IN_KM = 50.0
+DEFAULT_UNIT_OF_MEASUREMENT = 'km'
+
+SCAN_INTERVAL = timedelta(minutes=5)
+
+SIGNAL_DELETE_ENTITY = 'usgs_earthquake_hazards_program_feed_delete_{}'
+SIGNAL_UPDATE_ENTITY = 'usgs_earthquake_hazards_program_feed_update_{}'
+
+SOURCE = 'usgs_earthquake_hazards_program_feed'
+
+VALID_FEED_TYPES = [
+    'past_hour_significant_earthquakes',
+    'past_hour_m45_earthquakes',
+    'past_hour_m25_earthquakes',
+    'past_hour_m10_earthquakes',
+    'past_hour_all_earthquakes',
+    'past_day_significant_earthquakes',
+    'past_day_m45_earthquakes',
+    'past_day_m25_earthquakes',
+    'past_day_m10_earthquakes',
+    'past_day_all_earthquakes',
+    'past_week_significant_earthquakes',
+    'past_week_m45_earthquakes',
+    'past_week_m25_earthquakes',
+    'past_week_m10_earthquakes',
+    'past_week_all_earthquakes',
+    'past_month_significant_earthquakes',
+    'past_month_m45_earthquakes',
+    'past_month_m25_earthquakes',
+    'past_month_m10_earthquakes',
+    'past_month_all_earthquakes',
+]
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_FEED_TYPE): vol.In(VALID_FEED_TYPES),
+    vol.Optional(CONF_LATITUDE): cv.latitude,
+    vol.Optional(CONF_LONGITUDE): cv.longitude,
+    vol.Optional(CONF_RADIUS, default=DEFAULT_RADIUS_IN_KM): vol.Coerce(float),
+    vol.Optional(CONF_MINIMUM_MAGNITUDE, default=DEFAULT_MINIMUM_MAGNITUDE):
+        vol.All(vol.Coerce(float), vol.Range(min=0))
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the USGS Earthquake Hazards Program Feed platform."""
+    scan_interval = config.get(CONF_SCAN_INTERVAL, SCAN_INTERVAL)
+    feed_type = config.get(CONF_FEED_TYPE)
+    coordinates = (config.get(CONF_LATITUDE, hass.config.latitude),
+                   config.get(CONF_LONGITUDE, hass.config.longitude))
+    radius_in_km = config[CONF_RADIUS]
+    minimum_magnitude = config.get(CONF_MINIMUM_MAGNITUDE)
+    # Initialize the entity manager.
+    feed = UsgsEarthquakeHazardsProgramFeedEntityManager(
+        hass, add_entities, scan_interval, coordinates, feed_type,
+        radius_in_km, minimum_magnitude)
+
+    def start_feed_manager(event):
+        """Start feed manager."""
+        feed.startup()
+
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, start_feed_manager)
+
+
+class UsgsEarthquakeHazardsProgramFeedEntityManager:
```

Comment:
Can we find a shorter but still distinct name?

Reply:
I'm not a big fan of abbreviations; `Usgs` is already an abbreviation. `EarthquakeHazardsProgram` can probably be shortened; `FeedEntityManager` is trying to distinguish this class from the `FeedManager` that the library defines.
How about `UsgsEarthquakeFeedEntityManager`?

Reply:
Perhaps, yes. We try to name the component and platforms after the brands or organizations that provide them. So the new suggestion would still fit, right?

Reply:
The organisation providing the data is called "U.S. Geological Survey" or abbreviated to USGS. As part of their "Earthquake Hazard Program" they publish feeds about seismic activities. They also provide other data about water, landslides, volcanoes, droughts, wildfires, etc. I haven't looked at any of the others to assess their suitability for a Home Assistant integration. But it probably makes sense to have the ability the later on tap into more data sources from the USGS.

I'll rename to `UsgsEarthquakesFeedEntityManager` and `usgs_earthquakes_feed` then.